wagtail/embeds/embeds.py

```diff
@@ -40,13 +44,24 @@ def finder(url, max_width=None):
 
     # Create database record
     embed, created = Embed.objects.get_or_create(
-        url=url,
-        max_width=max_width,
-        defaults=embed_dict,
+        hash=embed_hash,
+        defaults=dict(
+            url=url,
+            max_width=max_width,
+            **embed_dict,
+        )
     )
 
     # Save
     embed.last_updated = datetime.now()
     embed.save()
 
     return embed
+
+
+def get_embed_hash(url, max_width=None):
+    h = md5()
+    h.update(url.encode("utf-8"))
+    if max_width is not None:
+        h.update(str(max_width).encode("utf-8"))
```

Comment:
Super-paranoid nitpick: putting the two `update`s together means that `get_embed_hash('https://example.com/embed', 400)` and `get_embed_hash('https://example.com/embed400')` return the same result. Easy enough to fix - I'll just add a `\n` delimiter between them...